sound/soc/soc-topology.c

```diff
@@ -2052,8 +2052,8 @@ static struct snd_soc_dai_link *snd_soc_find_dai_link(struct snd_soc_card *card,
 		if (name && (!link->name || strcmp(name, link->name)))
 			continue;
 
-		if (stream_name && (!link->stream_name
-				    || strcmp(stream_name, link->stream_name)))
+		if (stream_name && (!link->stream_name ||
+				    !strstr(link->stream_name, stream_name)))
```

Comment:
how about input with "playback" or "Capture", will it lead to unexpected status? is this condition too “generous”？

Reply:
@btian1 have you seen such names in our topologies? A stream name is usually indicative of the type of DAI and its dai index. To  give the widget a stream_name or DAI link stream name like "Playback" would be shooting yourself in the foot even with the original match

Reply:
soc-topology.c is not only for 'our topologies', extra care must be done to not break other users.

Reply:
Observation after a quick test locally: the `stream_name` is **always** NULL..
```
# dmesg | grep -e "Topology file" -e PETER
[ 2024.323723] snd_sof:sof_print_profile_info: sof-audio-pci-intel-tgl 0000:00:1f.3:  Topology file:     intel/avs-tplg/sof-hda-generic.tplg
[ 2024.472508] [PETER] snd_soc_find_dai_link: name: iDisp1
[ 2024.472510] [PETER] snd_soc_find_dai_link: link->name: iDisp1
[ 2024.472512] [PETER] snd_soc_find_dai_link: stream_name: (null)
[ 2024.472514] [PETER] snd_soc_find_dai_link: link->stream_name: iDisp1
[ 2024.472521] [PETER] snd_soc_find_dai_link: name: iDisp2
[ 2024.472523] [PETER] snd_soc_find_dai_link: link->name: iDisp2
[ 2024.472524] [PETER] snd_soc_find_dai_link: stream_name: (null)
[ 2024.472526] [PETER] snd_soc_find_dai_link: link->stream_name: iDisp2
[ 2024.472532] [PETER] snd_soc_find_dai_link: name: iDisp3
[ 2024.472533] [PETER] snd_soc_find_dai_link: link->name: iDisp3
[ 2024.472535] [PETER] snd_soc_find_dai_link: stream_name: (null)
[ 2024.472536] [PETER] snd_soc_find_dai_link: link->stream_name: iDisp3
[ 2024.472542] [PETER] snd_soc_find_dai_link: name: Analog Playback and Capture
[ 2024.472543] [PETER] snd_soc_find_dai_link: link->name: Analog Playback and Capture
[ 2024.472545] [PETER] snd_soc_find_dai_link: stream_name: (null)
[ 2024.472547] [PETER] snd_soc_find_dai_link: link->stream_name: Analog Playback and Capture
```
with:
```diff
diff --git a/sound/soc/soc-topology.c b/sound/soc/soc-topology.c
index 47ab5cf99497..8d898f18876c 100644
--- a/sound/soc/soc-topology.c
+++ b/sound/soc/soc-topology.c
@@ -2049,9 +2049,15 @@ static struct snd_soc_dai_link *snd_soc_find_dai_link(struct snd_soc_card *card,
 		if (link->id != id)
 			continue;
 
+		pr_warn("[PETER] %s: name: %s\n", __func__, name);
+		pr_warn("[PETER] %s: link->name: %s\n", __func__, link->name);
+
 		if (name && (!link->name || strcmp(name, link->name)))
 			continue;
 
+		pr_warn("[PETER] %s: stream_name: %s\n", __func__, stream_name);
+		pr_warn("[PETER] %s: link->stream_name: %s\n", __func__, link->stream_name);
+
 		if (stream_name && (!link->stream_name
 				    || strcmp(stream_name, link->stream_name)))
 			continue;
```

Reply:
> The change is NOP since it will never be executed?

@ujfalusi dai_link->stream_name will be given the same name as its name in soc_check_tplg_fes()

Reply:
closing this thread. I've also added partial matching to the link name to fix the issue @ujfalusi mentioned.